### PR TITLE
Backport fetcher thread leaking to 0.6 stable

### PR DESCRIPTION
**Pull request by nguyenquangminh0711**
Backport the fix https://github.com/zendesk/ruby-kafka/pull/654 https://github.com/zendesk/ruby-kafka/pull/634 https://github.com/zendesk/ruby-kafka/pull/644 https://github.com/zendesk/ruby-kafka/pull/641  to version 0.6 stable

### CODE REVIEW

File: lib/kafka/datadog.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def statsd=(statsd)
       end
 
       def host
-        @host ||= ::Datadog::Statsd::DEFAULT_HOST
+        @host ||= ::Kafka::Statsd::DEFAULT_HOST
```
Comment (tjwp) [inline lib/kafka/datadog.rb]: This change wasn't part of #654 and will break compatibility with dogstatsd-ruby < 4.0. See https://github.com/zendesk/ruby-kafka/pull/634/files.

Comment (nguyenquangminh0711) [reply lib/kafka/datadog.rb]: I see. So, I'll cherry pick the changes in that PRs to backport to 0.6.x too.

**Comment by tjwp** [comment]
@nguyenquangminh0711 Great! Thanks for picking up those other fixes.

**Comment by nguyenquangminh0711** [comment]
Done, @dasch please have a look and release a new version for v0.6.x. Thank you ❤️

**Comment by tjwp** [comment]
Attempted to update the changelog for a new release of v0.6.x here: https://github.com/zendesk/ruby-kafka/pull/671

**Pull request merged**